### PR TITLE
dal2: Add s3 support layout

### PR DESCRIPTION
**Pull request by Xuanwo**
Signed-off-by: Xuanwo <github@xuanwo.io>

I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Part of https://github.com/datafuselabs/databend/issues/3774

## Changelog

- New Feature

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

**Comment by Xuanwo** [comment]
@dantengsky @Veeupup I found iplement s3 support is more complex so I spilt it into #3774 instead. PTAL, thanks!

File: common/dal2/Cargo.toml [hunk 1]
```diff
@@ -15,8 +15,11 @@ async-compat = "0.2.1"
...
 bytes = "1"
 futures = "0.3.19"
-rusoto_core = "0.47.0"
-rusoto_s3 = "0.47.0"
-rusoto_credential = "0.47.0"
 tokio = { version = "1.15.0", features = ["fs"] }
 thiserror = "1"
+aws-config = "0.3.0"
+aws-types = {version = "0.3.0", features = ["hardcoded-credentials"]}
+aws-sdk-s3 = "0.3.0"
```
Comment (dantengsky) [inline common/dal2/Cargo.toml:22]: Hurrah!

File: common/dal2/src/services/s3.rs [hunk 2]
```diff
@@ -12,64 +12,202 @@
...
+        self
+    }
+
+    pub fn disable_ssl(&mut self) -> &mut Self {
+        self.disable_ssl = true;
+
+        self
+    }
+
+    pub fn enable_path_style(&mut self) -> &mut Self {
```
Comment (dantengsky) [inline common/dal2/src/services/s3.rs:76]: not sure if the signature of method `enable_path_style` is composable with the `finish` method, e.g. `builder.enable_path_style().finish()?`

Comment (dantengsky) [reply common/dal2/src/services/s3.rs:76]: even if they are not composable, lets' fix it in #3774

**Review by PsiACE — APPROVED** [review]

File: common/dal2/src/services/s3.rs (hunk 2 above)
Comment (Xuanwo) [reply common/dal2/src/services/s3.rs:76]: Quoted inside https://github.com/datafuselabs/databend/issues/3774

**Pull request merged**